src/coreclr/jit/reglist.h

```diff
@@ -9,7 +9,7 @@
 
 // The "regList" type is a small set of registerse
 #ifdef TARGET_X86
-typedef TinyArray<unsigned short, regNumber, REGNUM_BITS> regList;
+typedef TinyArray<unsigned int, regNumber, REGNUM_BITS> regList;
```

Comment:
Why this change?

Reply:
It caused by `ebx` usage for `VASigCookie` passing to `GenericPInvokeCalliHelper`. `ebx` is a fourth register (eax, ecx, edx, ebx) and `REGNUM_BITS = 6`: it leads to overflow of `unsigned short` type. This overflow appears via assertion:
https://github.com/dotnet/runtime/blob/9703660baa08914773b26e413e361c8ce04e6d94/src/coreclr/jit/gentree.cpp#L12334
Right part was 0 for `ebx` due to this overflow.

Reply:
Ok. Seems like some assert in tinyarray.h should have fired with this overflow.

Reply:
Thanks, I have added a fix into assert in tinyarray.h.

Reply:
Thanks, I've fixed assert in tinyarray.h.